flink-python/src/main/java/org/apache/flink/api/python/PythonGatewayServer.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.python;
+
+import py4j.GatewayServer;
+
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.InetAddress;
+
+/**
+ * The Py4j Gateway Server provides RPC service for user's python process.
+ */
+public class PythonGatewayServer {
```

Comment:
I think It's better to move this class into `flin-core` module, due to `flink-python` only conatains python scripts, and Eliminate the interdependence of flink-table and flink-python (future). And create a root package for python in `flink-core`, such as : `org.apache.flink.api.python`.  What do you think?

Reply:
Agreed. we should consider this potential dependency problem.